homeassistant/components/owfs.py

```diff
@@ -0,0 +1,468 @@
+    mod = hass.data[DATA_OWFS]
+
+    if config:
+        assert not discovery_info, discovery_info
+        dev = mod.service.get_device(config[CONF_ADDRESS])
+
+        attr = config.get(CONF_ATTRIBUTE)
+        if attr is not None:
+            cls = UnknownDevice
+        else:
+            try:
+                cls = classes[dev.family]
+            except KeyError:
+                cls = UnknownDevice
+
+        obj = cls(hass, dev, config=config)
+        await obj._init(False)
+
+        poll = config.get(CONF_POLL, {})
+        for k, v in poll.items():
+            if k == 'attr':
+                obj._do_poll = v
+            else:
+                await dev.set_polling_interval(k, v)
+
+        async_add_entities((obj, ))
+
+    else:
+        objs = []
+        for dev in discovery_info.get('devices', ()):
+            try:
+                dev._OWFSDevice__objects['_default']
+            except (KeyError, AttributeError):
+                pass
+            else:
+                continue
+
+            try:
+                cls = classes[dev.family]
+            except KeyError:
+                cls = UnknownDevice
+            obj = cls(hass, dev)
+            await obj._init(True)
+            objs.append(obj)
+        if objs:
+            async_add_entities(objs)
+
+
+class OWFSDevice(Entity):
+    """Representation of (an attribute of an) OWFS device"""
+    _attr = None
+    _val = None
+    _do_poll = 0
+    _next_poll = 0
+    _events = None  # list: remembered events
+
+    def __init__(self, hass: HomeAssistant, dev: "trio_owfs.device.Device",
+                 config: dict = {}):
+        self.hass = hass
+        self.dev = dev
+        self._events = None
+
+        name = config.get(CONF_NAME)
+        if name is None:
+            name = "{} {}".format(self.__class__.__name__, dev.id)
+        self._name = name
+        self._config = config
+
+        # save a ref to this device
+        try:
+            reg = dev.__objects
+        except AttributeError:
+            dev.__objects = reg = WeakValueDictionary()
+        attr = config.get(CONF_ATTRIBUTE)
+        if attr is None:
+            attr = "_default"
+        else:
+            self._attr = attr
+        reg[attr] = self
+
+    async def _init(self, without_config: bool):
+        """Async part of device initialization.
+        This code processes queued events and ensures that no more events
+        may be queued."""
+        events, self._events = self._events, None
+        for e in events:
+            await self.process_event(e)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def should_poll(self):
+        """overrides this Entity property.
+        OWFS doesn't require polling by HASS.
+        """
+        if not self._do_poll:
+            return False
+        t = time()
+        if self._next_poll > t:
+            return False
+        self._next_poll = t + self._do_poll
```

Comment:
Home Assistant only checks this property once. This doesn't work. If you want to create your own polling interval, call `self.async_schedule_update_ha_state()` when it's time for an update.